### PR TITLE
For #6548: Force expand toolbar if touch unhandled

### PR DESCRIPTION
**Pull request by sblatz**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBottomBehavior.kt [hunk 1]
```diff
@@ -75,6 +75,12 @@ class BrowserToolbarBottomBehavior(
         axes: Int,
         type: Int
     ): Boolean {
+        // Force expand the toolbar if event is unhandled, otherwise user could get stuck in a
+        // state where they cannot show the toolbar
+        if (engineView?.getInputResult() == EngineView.InputResult.INPUT_RESULT_UNHANDLED) {
+            forceExpand(child)
```
[the remark this answers is not in the document]
Comment (sblatz) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBottomBehavior.kt:81]: @Mugurell thanks for the suggestions! I'll update the PR 😄

[the remark this answers is not in the document]
Comment (sblatz) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBottomBehavior.kt]: @Mugurell how's that change? :)

[the remark this answers is not in the document]
Comment (Mugurell) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBottomBehavior.kt]: LGreatTM!
Thanks!

**Comment by sblatz** [comment]
bors r=mugurell

**Comment by sblatz** [comment]
bors retry

**Pull request merged**